### PR TITLE
Respect query limit in export sidebar

### PR DESCRIPTION
**Pull request by paescuj**
Before this change the user was able to set a limit higher than the current query limit which then results into an error on export. Now the export location will be locked to the File Library if the limit exceeds the query limit.

### CODE REVIEW

**Review by jaads — APPROVED** [review]
the value of the limit input can now not exceed the `QUERY_LIMIT_MAX` value and hence the limit error cannot be thrown anymore ✅

File: app/src/views/private/components/export-sidebar-detail.vue [hunk 1]
```diff
@@ -114,7 +114,14 @@
...
 				<div class="field half-right">
 					<p class="type-label">{{ t('limit') }}</p>
-					<v-input v-model="exportSettings.limit" type="number" :min="-1" :step="1" :placeholder="t('unlimited')" />
+					<v-input
+						v-model="exportSettings.limit"
+						type="number"
+						:min="queryLimitMax === Infinity ? -1 : 0"
+						:max="queryLimitMax"
+						:step="1"
+						:placeholder="queryLimitMax === Infinity && t('unlimited')"
```
Comment (licitdev) [inline app/src/views/private/components/export-sidebar-detail.vue:123]: Noticed a warning being thrown for the `placeholder` whenever the limit is changed.
 
image

**Review by licitdev — CHANGES_REQUESTED** [review]
Should the export be "lockedToFiles" when the query limit is exceeded, similar to large exports? 🤔 

https://github.com/directus/directus/blob/28d0228052da72055fbe8c8892c2c4947e0d9bbe/app/src/views/private/components/export-sidebar-detail.vue#L355-L358

File: app/src/views/private/components/export-sidebar-detail.vue (hunk 1 above)
Comment (rijkvanzanten) [inline app/src/views/private/components/export-sidebar-detail.vue:123]:
```suggestion
						:placeholder="queryLimitMax === Infinity ? t('unlimited') : undefined"
```

File: app/src/lang/translations/en-US.yaml [hunk 2]
```diff
@@ -463,13 +463,16 @@ open_file_in_tab: Open file in new tab
...
-exporting_no_items_to_export: No items to export. Adjust the exporting configuration.
+exporting_limited_items_in_collection: Exporting {count} out of {total} items within {collection}.
+exporting_no_items_to_export: No items to export. Adjust the exporting configuration and/or add items to the collection.
 exporting_download_hint: Once completed, this {format} file will automatically be downloaded to your device.
 exporting_batch_hint: This export will be processed in batches, and once completed, the {format} file will be saved to the File Library.
 exporting_batch_hint_forced:
   Due to the large number of items, this export must be processed in batches, and once completed, the {format} file will
   be saved to the File Library.
+exporting_query_limit_hint_forced:
+  Due to the number of items exceeding the query limit, this export must be processed in batches, and once completed, the {format} file will
```
Comment (jaads) [inline app/src/lang/translations/en-US.yaml:474]: for my taste the sentence is a bit too long.
 ```suggestion
  Due to the number of items exceeding the query limit, this export must be processed in batches. Once completed, the {format} file will
```

**Review by jaads — APPROVED** [review]
Also after the rework, the exception is still gone. Instead the file indeed goes to the file lib when the max query limit is exceeded. The Code looks good to me but the messages can probably be improved a bit 🤔

File: app/src/lang/translations/en-US.yaml (hunk 2 above)
Comment (paescuj) [reply app/src/lang/translations/en-US.yaml:474]: Also wonder if there is a better way to explain to the end-user that this is due to the query limit.
Perhaps @phazonoverload could help with the wording of these sentences? 😇

File: app/src/views/private/components/export-sidebar-detail.vue [hunk 3]
```diff
@@ -386,14 +377,50 @@ const getItemCount = debounce(async () => {
...
+		lockedToFiles.value = {
+			lockedBy: queryLimitThreshold ? 'queryLimitThreshold' : 'batchThreshold',
+			previousLocation: lockedToFiles.value?.previousLocation ?? location.value,
+		};
+
+		location.value = 'files';
+	} else if (lockedToFiles.value) {
+		location.value = lockedToFiles.value.previousLocation;
+		lockedToFiles.value = null;
 	}
```
Comment (licitdev) [inline app/src/views/private/components/export-sidebar-detail.vue:424]: This logic should also run upon load to correctly set the initial location and disabling download.

File: app/src/lang/translations/en-US.yaml (hunk 2 above)
Comment (rijkvanzanten) [reply app/src/lang/translations/en-US.yaml:474]: @paescuj I don't think we have to explain in-app _why_ it's done in batches

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: app/src/lang/translations/en-US.yaml (hunk 2 above)
Comment (paescuj) [reply app/src/lang/translations/en-US.yaml]: Simplified to:
https://github.com/directus/directus/blob/3cb97c3d91aef0e46f4f75a472faa1e0657f51bc/app/src/lang/translations/en-US.yaml#L473-L475

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**